File: app/src/main/java/io/github/muntashirakon/AppManager/db/Converters.java
// SPDX-License-Identifier: GPL-3.0-or-later

package io.github.muntashirakon.AppManager.db;

import androidx.room.TypeConverter;

public class Converters {
    @TypeConverter
    public static boolean fromInt(int value) {
        return value != 0;
    }

    @TypeConverter
    public static int toInt(boolean value) {
        return value ? 1 : 0;
    }

    @TypeConverter
    public static Boolean fromLong(Long value) {
        return value == null ? null : value != 0L;
    }

    @TypeConverter
    public static Long toLong(Boolean value) {
        return value == null ? null : (value ? 1L : 0L);
    }
}

File: app/src/main/java/io/github/muntashirakon/AppManager/db/Converters.kt
// SPDX-License-Identifier: GPL-3.0-or-later

package io.github.muntashirakon.AppManager.db

import androidx.room.TypeConverter

class Converters {
    @TypeConverter
    fun fromInt(value: Int): Boolean {
        return value != 0
    }

    @TypeConverter
    fun toInt(value: Boolean): Int {
        return if (value) 1 else 0
    }

    @TypeConverter
    fun fromLong(value: Long?): Boolean? {
        return if (value == null) null else value != 0L
    }

    @TypeConverter
    fun toLong(value: Boolean?): Long? {
        return if (value == null) null else (if (value) 1L else 0L)
    }
}
